src/Sarif.Converters/ConverterFactory.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    internal static class ConverterFactory
+    {
+        private static readonly IDictionary<string, Lazy<ToolFileConverterBase>> BuiltInConverters = CreateBuiltInConverters();
+
+        internal static ToolFileConverterBase CreateConverter(string toolFormat, string pluginAssemblyPath)
+        {
+            return string.IsNullOrWhiteSpace(pluginAssemblyPath)
+                ? CreateBuiltInConverter(toolFormat)
+                : CreateConverterFromPlugin(toolFormat, pluginAssemblyPath);
+        }
+
+        private static ToolFileConverterBase CreateBuiltInConverter(string toolFormat)
+        {
+            Lazy<ToolFileConverterBase> converter;
+            return BuiltInConverters.TryGetValue(toolFormat, out converter)
+                ? converter.Value
+                : null;
+        }
+
+        private static ToolFileConverterBase CreateConverterFromPlugin(string toolFormat, string pluginAssemblyPath)
+        {
+            if (!File.Exists(pluginAssemblyPath))
+            {
+                string message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    ConverterResources.ErrorMissingPluginAssembly,
+                    pluginAssemblyPath);
+
+                throw new ArgumentException(message, nameof(pluginAssemblyPath));
+            }
+
+            // Convention: The converter type name is derived from the tool name. It can
+            // reside in any namespace.
+            string converterTypeName = toolFormat + "Converter";
+
+            Assembly pluginAssembly = Assembly.LoadFile(pluginAssemblyPath);
```

Comment:
>LoadFile [](start = 47, length = 8)

You will run into problems here with plugins loaded in directory locations upstream, generally outside the driver exe install location. Converter dependencies may not be loaded properly from those locations. Make sense? So, someone has authored a converter that depends on a previously unloaded assembly alongside it. Won't be found. Resolution here is to enable a handler for unresolved assembly references and to use the directory of any loaded plug ins to attempt to resolve. ModernCop has some code for handling this i can point you to.